src/tests/unittests/utilities/test_tools.py

```diff
@@ -1,19 +1,34 @@
+import io
 import sys
+import time
 import unittest
-from unittest.mock import MagicMock, patch
 import warnings
+from contextlib import redirect_stdout
+from unittest.mock import MagicMock, patch
+
 import numpy as np
 import qcodes
 from qcodes.data.data_array import DataArray
 from qcodes.tests.legacy.data_mocks import DataSet2D
-from qtt.utilities import tools
-from qtt.utilities.tools import resampleImage, diffImage, diffImageSmooth, reshape_metadata, get_python_version, \
-    get_module_versions, get_git_versions, code_version, rdeprecated, in_ipynb, pythonVersion
+
 import qtt.measurements.scans
+from qtt.utilities import tools
```

Comment:
Do you want this if in the next line you import the single dependencies. You have to replace e.g. tools.code_version with code_version etc. and we get rid of the doubly imported stuff.

Reply:
Not part of the PR, but I removed it anyway